docs/proposals/2841-flux-hpc/README.md

```diff
@@ -0,0 +1,185 @@
+
+This approach provides an HPC environment without requiring the user to build a custom Docker image with Flux pre-installed, significantly improving the user experience.
+
+### API Design
+
+The proposed changes will integrate into the existing `v1alpha1` API structures. We will add a new field, `flux`, to the `MLPolicySource` struct. This aligns Flux with other runtimes like Torch and MPI.
+
+#### 1. `ClusterTrainingRuntime` and `TrainingRuntime`
+
+The proposal will leverage the existing `ClusterTrainingRuntime` (cluster-scoped) and `TrainingRuntime` (namespace-scoped) CRDs. These objects serve as reusable templates, and no changes are needed for their top-level structure.
+
+#### 2. Enhancing `MLPolicySource`
+
+We will add the `Flux` field to the `MLPolicySource` struct, which is embedded within the `MLPolicy`.
+
+```go
+type MLPolicySource struct {
+    Torch *TorchMLPolicySource `json:"torch,omitempty"`
+    MPI   *MPIMLPolicySource   `json:"mpi,omitempty"`
+
+    // FluxMLPolicy defines policy only for Flux
+    // +optional
+    Flux  *FluxMLPolicySource  `json:"flux,omitempty"`
+}
+```
+
+For the `FluxMLPolicySource`, we define the minimum required parameters needed for Flux and installing the view. The view must be compatible with the application container.
+
+```go
+// FluxMLPolicySource represents a Flux HPC runtime configuration.
+type FluxMLPolicySource struct {
+
+    // numNodes is the number of physical nodes for the job.
+    // This is defined a level up on the Trainer
+
+    // numProcPerNode is the number of processes per node.
+    // Defaults to 1.
+    // +kubebuilder:default=1
+    // +optional
+    NumProcPerNode *int32 `json:"numProcPerNode,omitempty"`
+
+    // Container image to use for Flux view that installs Flux
+    // This must be compatible with the application container
+    // Get the flux view container (these are choices)
+    // ghcr.io/converged-computing/flux-view-rocky:arm-9
+    // ghcr.io/converged-computing/flux-view-rocky:arn-8
+    // ghcr.io/converged-computing/flux-view-rocky:tag-9
+    // ghcr.io/converged-computing/flux-view-rocky:tag-8
+    // ghcr.io/converged-computing/flux-view-ubuntu:tag-noble
+    // ghcr.io/converged-computing/flux-view-ubuntu:tag-jammy
+    // ghcr.io/converged-computing/flux-view-ubuntu:tag-focal
+    // ghcr.io/converged-computing/flux-view-ubuntu:arm-jammy
+    // ghcr.io/converged-computing/flux-view-ubuntu:arm-focal
+    // We use an ubuntu (more recent) default since it is common
+    // +kubebuilder:default="ghcr.io/converged-computing/flux-view-ubuntu:arm-jammy"
+    Image string `json:"image,omitempty"`
+
+    // Network device for flux to use
+    // +kubebuilder:default="eth0"
+    NetworkDevice string `json:"networkDevice,omitempty"`
+
+    // Queue policy for Flux to use
+    // +kubebuilder:default="fcfs"
+    QueuePolicy string `json:"queuePolicy,omitempty"`
```

Comment:
Should this be enum?
Also, please can you explain a little bit how this queue works in Flux?
Is it part of: https://flux-framework.readthedocs.io/projects/flux-rfc/en/latest/spec_33.html ?

Reply:
Since we don't support long-living `TrainClusters` for now, we will bootstrap Flux and shutdown for every TrainJob submission. Which means we only submit single Job in the Flux cluster.

So I am trying to understand how this queue policy will be used?